[Android] Auto linker fails on new react-native 0.71.3 project

### What happened?

Created a new blank React Native project.
Installed react-native-navigation
Run the auto linker
Failed to modify the MainActivity.java & MainApplication.java to a working state

### What was the expected behaviour?

That the auto linker would correctly modify MainActivity.java & MainApplication.java to a working state

### Was it tested on latest react-native-navigation?

- [X] I have tested this issue on the latest react-native-navigation release and it still reproduces.

### Help us reproduce this issue!

Compile Issues
```
> Task :app:compileDebugJavaWithJavac FAILED
32 actionable tasks: 2 executed, 30 up-to-date
/Users/person/code/myapp/android/app/src/main/java/com/myapp/MainApplication.java:16: error: cannot find symbol
      new NavigationReactNativeHost(this) {
          ^
  symbol:   class NavigationReactNativeHost
  location: class MainApplication
/Users/person/code/myapp/android/app/src/main/java/com/myapp/MainApplication.java:17: error: method does not override or implement a method from a supertype
        @Override
        ^
/Users/person/code/myapp/android/app/src/main/java/com/myapp/MainApplication.java:22: error: method does not override or implement a method from a supertype
        @Override
        ^
/Users/person/code/myapp/android/app/src/main/java/com/myapp/MainApplication.java:25: error: no suitable constructor found for PackageList(<anonymous NavigationReactNativeHost>)
          List<ReactPackage> packages = new PackageList(this).getPackages();
                                        ^
    constructor PackageList.PackageList(ReactNativeHost) is not applicable
      (argument mismatch; <anonymous NavigationReactNativeHost> cannot be converted to ReactNativeHost)
    constructor PackageList.PackageList(Application) is not applicable
      (argument mismatch; <anonymous NavigationReactNativeHost> cannot be converted to Application)
/Users/person/code/myapp/android/app/src/main/java/com/myapp/MainApplication.java:31: error: method does not override or implement a method from a supertype
        @Override
        ^
/Users/person/code/myapp/android/app/src/main/java/com/myapp/MainApplication.java:36: error: method does not override or implement a method from a supertype
        @Override
        ^
/Users/person/code/myapp/android/app/src/main/java/com/myapp/MainApplication.java:41: error: method does not override or implement a method from a supertype
        @Override
        ^
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
7 errors


```

Here are my MainApplication & MainActivity. Any that can help correct configure them would be awesome.

MainApplication.java
```
package com.myapp;

import android.app.Application;
import com.facebook.react.PackageList;
import com.reactnativenavigation.NavigationApplication;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint;
import com.facebook.react.defaults.DefaultReactNativeHost;
import com.facebook.soloader.SoLoader;
import java.util.List;

public class MainApplication extends NavigationApplication {

  private final ReactNativeHost mReactNativeHost =
      new NavigationReactNativeHost(this) {
        @Override
        public boolean getUseDeveloperSupport() {
          return BuildConfig.DEBUG;
        }

        @Override
        protected List<ReactPackage> getPackages() {
          @SuppressWarnings("UnnecessaryLocalVariable")
          List<ReactPackage> packages = new PackageList(this).getPackages();
          // Packages that cannot be autolinked yet can be added manually here, for example:
          // packages.add(new MyReactNativePackage());
          return packages;
        }

        @Override
        protected String getJSMainModuleName() {
          return "index";
        }

        @Override
        protected boolean isNewArchEnabled() {
          return BuildConfig.IS_NEW_ARCHITECTURE_ENABLED;
        }

        @Override
        protected Boolean isHermesEnabled() {
          return BuildConfig.IS_HERMES_ENABLED;
        }
      };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();

    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
      // If you opted-in for the New Architecture, we load the native entry point for this app.
      DefaultNewArchitectureEntryPoint.load();
    }
    ReactNativeFlipper.initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
  }
}

```

MainActivity.java
```
package com.myapp;

import com.reactnativenavigation.NavigationActivity;
import com.facebook.react.ReactActivityDelegate;
import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint;
import com.facebook.react.defaults.DefaultReactActivityDelegate;

public class MainActivity extends NavigationActivity {



  /**
   * Returns the instance of the {@link ReactActivityDelegate}. Here we use a util class {@link
   * DefaultReactActivityDelegate} which allows you to easily enable Fabric and Concurrent React
   * (aka React 18) with two boolean flags.
   */
}
```

### In what environment did this happen?

React Native Navigation version: 7.32.1
React Native version: 0.71.3
Has Fabric (React Native's new rendering system) enabled: yes
Node version: 16.14.2
Device model: sdk_gphone64_arm64
Android version: 13

--- a/autolink/fixtures/rn71/AppDelegate.mm.template
+++ b/autolink/fixtures/rn71/AppDelegate.mm.template
@@ -1,0 +1,36 @@
+#import "AppDelegate.h"
+
+#import <React/RCTBundleURLProvider.h>
+
+@implementation AppDelegate
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
+{
+  self.moduleName = @"app";
+  // You can add your custom initial props in the dictionary below.
+  // They will be passed down to the ViewController used by React Native.
+  self.initialProps = @{};
+
+  return [super application:application didFinishLaunchingWithOptions:launchOptions];
+}
+
+- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
+{
+#if DEBUG
+  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index"];
+#else
+  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
+#endif
+}
+
+/// This method controls whether the `concurrentRoot`feature of React18 is turned on or off.
+///
+/// @see: https://reactjs.org/blog/2022/03/29/react-v18.html
+/// @note: This requires to be rendering on Fabric (i.e. on the New Architecture).
+/// @return: `true` if the `concurrentRoot` feature is enabled. Otherwise, it returns `false`.
+- (BOOL)concurrentRootEnabled
+{
+  return true;
+}
+
+@end

--- a/autolink/fixtures/rn71/MainActivity.java.template
+++ b/autolink/fixtures/rn71/MainActivity.java.template
@@ -1,0 +1,35 @@
+package com.app;
+
+import com.reactnativenavigation.NavigationActivity;
+import com.facebook.react.ReactActivityDelegate;
+import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint;
+import com.facebook.react.defaults.DefaultReactActivityDelegate;
+
+public class MainActivity extends NavigationActivity {
+
+  /**
+   * Returns the name of the main component registered from JavaScript. This is used to schedule
+   * rendering of the component.
+   */
+  @Override
+  protected String getMainComponentName() {
+    return "app";
+  }
+
+  /**
+   * Returns the instance of the {@link ReactActivityDelegate}. Here we use a util class {@link
+   * DefaultReactActivityDelegate} which allows you to easily enable Fabric and Concurrent React
+   * (aka React 18) with two boolean flags.
+   */
+  @Override
+  protected ReactActivityDelegate createReactActivityDelegate() {
+    return new DefaultReactActivityDelegate(
+        this,
+        getMainComponentName(),
+        // If you opted-in for the New Architecture, we enable the Fabric Renderer.
+        DefaultNewArchitectureEntryPoint.getFabricEnabled(), // fabricEnabled
+        // If you opted-in for the New Architecture, we enable Concurrent React (i.e. React 18).
+        DefaultNewArchitectureEntryPoint.getConcurrentReactEnabled() // concurrentRootEnabled
+        );
+  }
+}

--- a/autolink/postlink/activityLinker.js
+++ b/autolink/postlink/activityLinker.js
@@ -105,7 +105,7 @@ class ActivityLinker {
     if (this._hasCreateReactActivityDelegate(activityContent)) {
       debugn('   Removing createReactActivityDelegate function');
       return activityContent.replace(
-        /\/\*\*(\s+\*.+)+\s+@Override\s+protected ReactActivityDelegate createReactActivityDelegate\(\) {(\s*[\w*(,);])*\s*}/,
+        /\/\*\*\s*\n([^\*]|(\*(?!\/)))*\*\/\s*@Override\s*protected\s*ReactActivityDelegate\s*createReactActivityDelegate\s*\(\)\s*{\s*return((.|\r|\s)*?)}/,
         ''
       );
     } else {
@@ -115,7 +115,7 @@ class ActivityLinker {
   }
 
   _hasCreateReactActivityDelegate(activityContent) {
-    return /\/\*\*(\s+\*.+)+\s+@Override\s+protected ReactActivityDelegate createReactActivityDelegate\(\) {(\s*[\w*(,);])*\s*}/.test(
+    return /\/\*\*\s*\n([^\*]|(\*(?!\/)))*\*\/\s*@Override\s*protected\s*ReactActivityDelegate\s*createReactActivityDelegate\s*\(\)\s*{\s*return((.|\r|\s)*?)}/.test(
       activityContent
     );
   }

--- a/autolink/postlink/activityLinker.test.js
+++ b/autolink/postlink/activityLinker.test.js
@@ -74,4 +74,30 @@ describe('activityLinker', () => {
     const mainActivityContent = fs.readFileSync(linker.activityPath, 'utf8');
     expect(mainActivityContent).toMatchSnapshot();
   });
+
+  it('should work for RN 0.71', () => {
+    jest.mock('../postlink/path', () => {
+      const { copyFileSync } = require('fs');
+      const { tmpdir } = require('os');
+      const path = require('path');
+
+      const tmpMainActivityPath = path.resolve(tmpdir(), 'rnn-tests_MainActivity.java');
+
+      copyFileSync(
+        path.resolve('autolink/fixtures/rn71/MainActivity.java.template'),
+        tmpMainActivityPath
+      );
+
+      return {
+        mainActivityJava: tmpMainActivityPath,
+      };
+    });
+
+    const ActivityLinker = require('./activityLinker');
+    const linker = new ActivityLinker();
+
+    linker.link();
+    const mainActivityContent = fs.readFileSync(linker.activityPath, 'utf8');
+    expect(mainActivityContent).toMatchSnapshot();
+  });
 });

--- a/autolink/postlink/appDelegateLinker.js
+++ b/autolink/postlink/appDelegateLinker.js
@@ -100,12 +100,22 @@ class AppDelegateLinker {
       return content;
     }
 
-    debugn('   Bootstrapping Navigation');
-    return content.replace(
-      /RCTBridge.*];/,
-      'RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];\n' +
-        '[ReactNativeNavigation bootstrapWithBridge:bridge];'
-    );
+    debugn('   Bootstrapping Navigation !!!!');
+    return content
+      .replace(
+        /RCTBridge.*];/,
+        'RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];\n' +
+          '[ReactNativeNavigation bootstrapWithBridge:bridge];'
+      )
+      .replace(
+        /return \[super application:application didFinishLaunchingWithOptions:launchOptions\];/,
+        'return YES;'
+      )
+      .replace(
+        /self.moduleName.*;/,
+        'RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];\n' +
+          '  [ReactNativeNavigation bootstrapWithBridge:bridge];'
+      );
   }
 
   _doesBootstrapNavigation(content) {

--- a/autolink/postlink/appDelegateLinker.test.js
+++ b/autolink/postlink/appDelegateLinker.test.js
@@ -74,4 +74,30 @@ describe('appDelegateLinker', () => {
     const appDelegateContent = fs.readFileSync(linker.appDelegatePath, 'utf8');
     expect(appDelegateContent).toMatchSnapshot();
   });
+
+  it('should work for RN 0.71', () => {
+    jest.mock('../postlink/path', () => {
+      const { copyFileSync } = require('fs');
+      const { tmpdir } = require('os');
+      const path = require('path');
+
+      const tmpAppDelegatePath = path.resolve(tmpdir(), 'rnn-tests_AppDelegate.mm');
+
+      copyFileSync(
+        path.resolve('autolink/fixtures/rn71/AppDelegate.mm.template'),
+        tmpAppDelegatePath
+      );
+
+      return {
+        appDelegate: tmpAppDelegatePath,
+      };
+    });
+
+    const AppDelegateLinker = require('./appDelegateLinker');
+    const linker = new AppDelegateLinker();
+
+    linker.link();
+    const appDelegateContent = fs.readFileSync(linker.appDelegatePath, 'utf8');
+    expect(appDelegateContent).toMatchSnapshot();
+  });
 });

--- a/autolink/postlink/gradleLinker.js
+++ b/autolink/postlink/gradleLinker.js
@@ -3,7 +3,7 @@ var path = require('./path');
 var fs = require('fs');
 var { warnn, errorn, logn, infon, debugn } = require('./log');
 var { insertString } = require('./stringUtils');
-var DEFAULT_KOTLIN_VERSION = '1.4.31';
+var DEFAULT_KOTLIN_VERSION = '1.5.31';
 // This should be the minSdkVersion required for RNN.
 var DEFAULT_MIN_SDK_VERSION = 21;
 
@@ -80,7 +80,7 @@ class GradleLinker {
       return insertString(
         contents,
         match.index,
-        `classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:${DEFAULT_KOTLIN_VERSION}"\n        `
+        `classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$RNNKotlinVersion"\n        `
       );
     } else {
       throw new Error('   Could not add kotlin plugin dependency');

